android/java/org/chromium/chrome/browser/externalnav/BraveExternalNavigationHandler.java

```diff
@@ -29,7 +30,13 @@ public BraveExternalNavigationHandler(ExternalNavigationDelegate delegate) {
 
     @Override
     public OverrideUrlLoadingResult shouldOverrideUrlLoading(ExternalNavigationParams params) {
-        if (isWalletProviderOverride(params)) {
+        String originalUrl = params.getUrl().getSpec();
+        String referrerUrl = params.getReferrerUrl().getSpec();
+
+        Log.e("BraveExternalNavigationHandler",
+                "originalUrl : " + originalUrl + " referrerUrl : " + referrerUrl);
+        if (BraveRewardsNativeWorker.getInstance().isValidWalletProviderRedirect(
+                    +referrerUrl, originalUrl)) {
             String originalUrl = params.getUrl().getSpec();
             String url = originalUrl.replaceFirst("^rewards://", "brave://rewards/");
```

Comment:
Should use `brave_rewards::TransformUrl`.

Reply:
we need to remove the check we have here.